include/utilities/realtime/dataman/DataMan.h

```diff
@@ -0,0 +1,468 @@
+        if (dtype == "float")
+            return sizeof(float);
+        if (dtype == "double")
+            return sizeof(double);
+        if (dtype == "long double")
+            return sizeof(long double);
+        if (dtype == "complex<float>")
+            return sizeof(float) * 2;
+        if (dtype == "complex<double>")
+            return sizeof(double) * 2;
+
+        if (dtype == "int8_t")
+            return sizeof(int8_t);
+        if (dtype == "uint8_t")
+            return sizeof(uint8_t);
+        if (dtype == "int16_t")
+            return sizeof(int16_t);
+        if (dtype == "uint16_t")
+            return sizeof(uint16_t);
+        if (dtype == "int32_t")
+            return sizeof(int32_t);
+        if (dtype == "uint32_t")
+            return sizeof(uint32_t);
+        if (dtype == "int64_t")
+            return sizeof(int64_t);
+        if (dtype == "uint64_t")
+            return sizeof(uint64_t);
+        return 0;
+    }
+
+    inline json atoj(unsigned int *array)
+    {
+        json j;
+        if (array)
+        {
+            if (array[0] > 0)
+            {
+                j = {array[1]};
+                for (unsigned int i = 2; i <= array[0]; i++)
+                {
+                    j.insert(j.end(), array[i]);
+                }
+            }
+        }
+        return j;
+    }
+
+    inline string rmquote(string in) { return in.substr(1, in.length() - 2); }
+
+    inline bool isin(string a, json j)
+    {
+        for (unsigned int i = 0; i < j.size(); i++)
+        {
+            if (j[i] == a)
+                return true;
+        }
+        return false;
+    }
+
+    inline int closest(int v, json j, bool up)
+    {
+        int s = 100, k = 0, t;
+        for (unsigned int i = 0; i < j.size(); i++)
+        {
+            if (up)
+                t = j[i].get<int>() - v;
+            else
+                t = v - j[i].get<int>();
+            if (t >= 0 && t < s)
+            {
+                s = t;
+                k = i;
+            }
+        }
+        return k;
+    }
+
+    inline void check_shape(json &p_jmsg)
+    {
+        vector<size_t> varshape;
+        if (check_json(p_jmsg, {"varshape"}))
+        {
+            varshape = p_jmsg["varshape"].get<vector<size_t>>();
+        }
+        else
+        {
+            return;
+        }
+        if (p_jmsg["putshape"] == nullptr)
+        {
+            p_jmsg["putshape"] = varshape;
+        }
+        if (p_jmsg["offset"] == nullptr)
+        {
+            p_jmsg["offset"] = vector<size_t>(varshape.size(), 0);
+        }
+        p_jmsg["putbytes"] = product(p_jmsg["putshape"].get<vector<size_t>>(),
+                                     dsize(p_jmsg["dtype"].get<string>()));
+        p_jmsg["varbytes"] =
+            product(varshape, dsize(p_jmsg["dtype"].get<string>()));
+    }
+
+    inline shared_ptr<DataMan> get_man(string method)
+    {
+        string soname = "lib" + method + "man.so";
```

Comment:
This is just for run-time optimization. If something optimal can be loaded dynamically, then DataMan will use it. If not, then it should not affect how the basic static part of DataMan works. For example, whenever anything goes wrong, no matter it is because of the Cray environment or a wrongly built XxxMan.so, this function will simply return a nullptr, and then the main DataMan runtime optimizer will just ignore it and keep doing what is provided in the .a library.